scripts/docker_run

```diff
@@ -3,24 +3,31 @@
 set -o errexit
 set -o xtrace
 
-readonly USERNAME="$(id -u -n)"
+# The default user for a Docker container has uid 0 (root). To avoid creating
+# root-owned files in the build directory we tell Docker to use the current user
+# ID, if known.
+# See
+# https://github.com/googleapis/google-cloud-cpp/blob/a186208b79d900b4ec71c6f9df3acf7638f01dc6/ci/kokoro/docker/build.sh#L147-L152
+readonly DOCKER_UID="${UID:-0}"
+readonly DOCKER_GID="${GID:-0}"
+readonly DOCKER_USER="${USER:-root}"
 
 mkdir -p './bazel-cache'
 mkdir -p './cargo-cache'
 
 docker build \
   --tag=oak \
-  --build-arg="USERNAME=$USERNAME" \
-  --build-arg="UID=$(id -u)" \
-  --build-arg="GID=$(id -g)" \
   .
 
+# Keep this in sync with /scripts/run_examples .
 docker run \
   --interactive \
   --tty \
-  --volume="$PWD/bazel-cache:/home/$USERNAME/.cache/bazel" \
-  --volume="$PWD/cargo-cache:/home/$USERNAME/.cargo/registry" \
-  --volume="$PWD":/opt/my-project \
+  --user="$DOCKER_UID:$DOCKER_GID" \
+  --env="USER=$DOCKER_USER" \
```

Comment:
Why do we need this in the env?

Reply:
It's possible that I'm just cargo culting and that it may not be required. I'll ping @coryan for why Bazel might need this environment variable.

Reply:
I get this error message from Bazel when the variable is not set:

```
FATAL: $USER is not set, and unable to look up name of current user: (error: 0): Success
```

Reply:
Also, you may want to consider using an array so you can document why the options are there, for example:

https://github.com/googleapis/google-cloud-cpp-spanner/blob/ad53c95a8ff7e9aa16396f3ae0f5cd5be31bb25b/ci/kokoro/docker/build.sh#L170

Reply:
Indeed, I had to explicitly add it because I got that error when I first tried. It seems quite odd for bazel to have this dependency, but we are stuck with it for now. And yes, I wanted to document this but it's not possible to do that inline to this command :/

Reply:
Hey! I am running into issues with the user permissions. 

The container is run with `--user=2000:0 --env=USER=travis` but there is not user with id 2000 in the container. As soon as its accessing local files, like the cargo lock file, it generates a  user permission errors.

https://github.com/project-oak/oak/pull/112#issuecomment-504936861